extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/SonosHandlerFactory.java

```diff
@@ -95,19 +94,39 @@ protected ThingHandler createHandler(Thing thing) {
             ZonePlayerHandler handler = new ZonePlayerHandler(thing, upnpIOService, discoveryServiceRegistry, opmlUrl);
 
             // register the speaker as an audio sink
-            SonosAudioSink audioSink = new SonosAudioSink(context, handler, audioHTTPServer, callbackUrl);
+            String callbackUrl = createCallbackUrl();
+            SonosAudioSink audioSink = new SonosAudioSink(handler, audioHTTPServer, callbackUrl);
```

Comment:
Does a configuration change and so a re-activate trigger a recreation of all thing handlers, too?